src/clients/QBittorrent.ts

```diff
@@ -244,39 +240,55 @@ export default class QBittorrent implements TorrentClient {
 	 */
 	async generateCorrectSavePath(
 		searchee: Searchee,
-		{ save_path },
-	): Promise<{ save_path: string }> {
-		const subfolderContentLayout =
-			await this.isSubfolderContentLayout(searchee);
+		torrentInfo: TorrentInfo,
+	): Promise<string> {
+		const subfolderContentLayout = await this.isSubfolderContentLayout(
+			searchee,
+			torrentInfo,
+		);
 		if (subfolderContentLayout) {
-			return {
-				save_path: join(save_path, stripExtension(searchee.name)),
-			};
+			return dirname(torrentInfo.content_path);
 		}
-		return { save_path };
+		return torrentInfo.save_path;
+	}
+
+	/*
+	@return array of query results
+	 */
+	async getAllTorrentInfo(): Promise<TorrentInfo[]> {
+		const responseText = await this.request("/torrents/info", "");
+		return JSON.parse(responseText);
+	}
+
+	/*
+	@param hash the hash of the torrent or undefined for all torrents
+	@return array of query results
+	 */
+	async getTorrentInfo(hash: string): Promise<TorrentInfo[]> {
+		const torrents = await this.getAllTorrentInfo();
+		return torrents.filter(
+			(torrent) =>
+				hash === torrent.hash ||
+				hash === torrent.infohash_v1 ||
+				hash === torrent.infohash_v2,
```

Comment:
is v2 infohash ever the same as v1?

our searchee's don't presently support v2 and it seems a long way off from adoption that would be relevant to us let?